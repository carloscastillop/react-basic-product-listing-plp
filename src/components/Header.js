import React from 'react';
import '../styles/App.scss';

class Header extends React.Component {
    render() {
        return (
            <React.Fragment>
                <header
                    className="my-3"
                    data-testid="header"
                >
                    <div className="container">
                        <nav className="navbar navbar-light bg-light">
                            <span className="navbar-brand mb-0 h1">
                                <i className="fas fa-tshirt"></i> Clothing
                            </span>
                        </nav>
                    </div>
                </header>
            </React.Fragment>
        )
    }
}

export default Header;