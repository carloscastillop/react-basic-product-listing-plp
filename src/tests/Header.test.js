import React from 'react';
import ReactDOM from 'react-dom';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

import Header from '../components/Header';

afterEach(cleanup);

/**
 * Basic render
 */
it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header/>, div);
});

/**
 * Basic render 2
 */
it('renders header', () => {
    const {getByTestId} = render(<Header/>)
    expect(getByTestId('header')).toHaveTextContent('Clothing');
});

/**
 * Snapshot
 */
it('matches snapshot', () => {
    const tree = renderer.create(<Header/>).toJSON();
    expect(tree).toMatchSnapshot();
});