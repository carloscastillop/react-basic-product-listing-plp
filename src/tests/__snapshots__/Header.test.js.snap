// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<header
  className="my-3"
  data-testid="header"
>
  <div
    className="container"
  >
    <nav
      className="navbar navbar-light bg-light"
    >
      <span
        className="navbar-brand mb-0 h1"
      >
        <i
          className="fas fa-tshirt"
        />
         Clothing
      </span>
    </nav>
  </div>
</header>
`;
